mdl/src/main/herd/scripts/installDBForHerd.sh

```diff
@@ -56,22 +56,40 @@ fi
+
+# determine if a new Herd version has been requested for rolling-deployment purposes
+requestedVersion=$(aws ssm get-parameter --name /app/MDL/${mdlInstanceName}/${environment}/HERD/RequestedVersion --region ${region} --output text --query Parameter.Value)
+
+# determine if this is a rolling-deployment
+herdRollingDeployment=$(aws ssm get-parameter --name /app/MDL/${mdlInstanceName}/${environment}/HERD/DeploymentInvoked --region ${region} --output text --query Parameter.Value)
+
 # Execute DB scripts
 herdDatabasePassword=$(aws ssm get-parameter --name /app/MDL/${mdlInstanceName}/${environment}/HERD/RDS/masterAccount --with-decryption --region ${region} --output text --query Parameter.Value)
-if [ "${herdDatabasePassword}" = "changeit" ] ; then
-    # changing DB password
-    herdDatabasePassword=$(openssl rand -base64 32 | tr -d /=+ | cut -c -16)
-    aws ssm put-parameter --name /app/MDL/${mdlInstanceName}/${environment}/HERD/RDS/masterAccount --type SecureString --value ${herdDatabasePassword} --region ${region} --overwrite
-    check_error $? "aws ssm put-parameter --name '/app/MDL/${mdlInstanceName}/${environment}/HERD/RDS/masterAccount' secure string"
+
+echo "Deployment invoked: ${herdRollingDeployment}"
+
+if [ "${herdRollingDeployment}" = "false" ] ; then
+
+    if [ "${herdDatabasePassword}" = "changeit" ] ; then
+        # change DB password
+        herdDatabasePassword=$(openssl rand -base64 32 | tr -d /=+ | cut -c -16)
+        aws ssm put-parameter --name /app/MDL/${mdlInstanceName}/${environment}/HERD/RDS/masterAccount --type SecureString --value ${herdDatabasePassword} --region ${region} --overwrite
+        check_error $? "aws ssm put-parameter --name '/app/MDL/${mdlInstanceName}/${environment}/HERD/RDS/masterAccount' secure string"
+    fi
+
 fi
 
-# Change RDS password to the value in parameter store
-aws rds modify-db-instance --db-instance-identifier ${herdDBInstance} --master-user-password ${herdDatabasePassword} --apply-immediately --region ${region}
-check_error $? "aws rds modify-db-instance --db-instance-identifier ${herdDBInstance} modify password"
-#sleep 2 minutes to wait for the rds status changed to resetting-master-password
-sleep 180
-# Waiting for the new password to take effect, which is waiting until rds is available again
-execute_cmd "aws rds wait db-instance-available --db-instance-identifier ${herdDBInstance} --region ${region}"
+if [ "${herdRollingDeployment}" = "false" ] ; then
+    # Change RDS password to the value in parameter store (only do this if not a rolling deployment)
+    aws rds modify-db-instance --db-instance-identifier ${herdDBInstance} --master-user-password ${herdDatabasePassword} --apply-immediately --region ${region}
+    check_error $? "aws rds modify-db-instance --db-instance-identifier ${herdDBInstance} modify password"
+    #sleep 2 minutes to wait for the rds status to change to resetting-master-password
```

Comment:
The comment says 2 minutes, but the sleep is for 3 minutes
```suggestion
    #sleep 3 minutes to wait for the rds status to change to resetting-master-password
```